src/shared/config/AuthXmlConfig.php

```diff
@@ -0,0 +1,35 @@
+<?php declare(strict_types = 1);
+namespace PharIo\Phive;
+
+class AuthXmlConfig implements AuthConfig {
+    /** @var XmlFile */
+    private $xmlFile;
+
+    /**
+     * AuthXmlConfig constructor.
+     */
+    public function __construct(XmlFile $xmlFile) {
+        $this->xmlFile = $xmlFile;
+    }
+
+    public function hasAuthentication(string $domain): bool {
+        $query  = \sprintf('//phive:domain[@host="%s"]', $domain);
+        $result = $this->xmlFile->query($query);
+
+        return $result->count() > 0;
+    }
+
+    public function getAuthentication(string $domain): Authentication {
+        $query  = \sprintf('//phive:domain[@host="%s"]', $domain);
+        $result = $this->xmlFile->query($query);
+
+        /** @var \DOMElement $auth */
+        $auth = $result->item(0);
+
+        if (\mb_strtolower($auth->getAttribute('type')) === 'basic' && $auth->hasAttribute('username')) {
```

Comment:
This will produce an invalid Authentication in case the `type` is set to `basic` but no username is given. Or the `password` attribute is missing.

This full passage is fragile.

Reply:
My assumption that user will fill every field correctly doesn't match the "Never trust user input" adage 😅 
I will refactor this whole (type="basic") part into its own dedicate method to ease readability